scripts/metric_reporter/parser/junit_xml_parser.py

```diff
@@ -5,85 +5,263 @@
 """Module for parsing test suite results from JUnit XML content."""
 
 import logging
-import re
 from pathlib import Path
 from typing import Any
 
-import defusedxml.ElementTree as ElementTree
+import xmltodict
 from pydantic import BaseModel, ValidationError
 
 from scripts.metric_reporter.parser.base_parser import ParserError, JOB_DIRECTORY_PATTERN
 
 
-class JUnitXmlProperty(BaseModel):
-    """Represents a property of a test case."""
+class JestJUnitXmlTestCase(BaseModel):
+    """Represents a test case in a test suite."""
 
     name: str
-    value: str
+    classname: str
+    time: float
+    skipped: str | None = None
+    failure: str | None = None
 
 
-class JUnitXmlSkipped(BaseModel):
-    """Represents a skipped test case."""
+class JestJUnitXmlTestSuite(BaseModel):
+    """Represents a test suite containing multiple test cases."""
+
+    name: str
+    timestamp: str
+    tests: int
+    failures: int
+    skipped: int
+    time: float
+    errors: int
+    test_cases: list[JestJUnitXmlTestCase]
```

Comment:
There does seem to be some repetition here, so I wonder if we could have a base class we use to expand from, but no big deal.

Reply:
They are indeed very similar, which is why the first approach was to use objects that accommodate all 6 Junit XML versions. When you overlap the 6 versions, there is very little crossover, mostly because of TAP. IMO adding that depth of inheritance would add complexity without much benefit at this time. If we can stop supporting TAP though, this might change.